internal/analytics/deferred.go

```diff
@@ -0,0 +1,65 @@
+package analytics
+
+import (
+	"encoding/json"
+
+	"github.com/spf13/viper"
+
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+)
+
+var Defer bool
+
+type deferredData struct {
+	Category   string
+	Action     string
+	Label      string
+	Dimensions map[string]string
+}
+
+const deferredCfgKey = "deferrer_analytics"
+
+func deferEvent(category, action, label string, dimensions map[string]string) error {
+	logging.Debug("Deferring: %s, %s, %s", category, action, label)
+	deferred := loadDeferred()
+	deferred = append(deferred, deferredData{category, action, label, dimensions})
+	saveDeferred(deferred)
+	viper.WriteConfig() // the global viper instance is bugged, need to work around it for now -- https://www.pivotaltracker.com/story/show/175624789
```

Comment:
This is the write I mentioned in the other thread; The potential error is not handled.